src/lib.rs

```diff
@@ -211,7 +219,7 @@ impl Ord for Signature {
 
 impl Distribution<Signature> for Standard {
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Signature {
-        Signature(rng.gen04())
+        Signature(G2::random(&mut ChaChaRng::from_rng(rng).unwrap()))
```

Comment:
Why not use `rng` itself?

Reply:
This is the only point that isn't clear to me yet: Is there a reason why we shouldn't just use `rng` itself here?